### ISSUE TITLE
Clarify the allowed uses of "Mastodon" trademark for the instances

### ISSUE THREAD

**Issue by collimarco** [mastodon/mastodon#20771]
### Pitch

I cannot find enough information about the legal, allowed use of the name "Mastodon" for the instances (i.e. independent websites that run an instance of this software).

For example, can you run a domain that has the word "mastodon" in its name? For example, "mastodon.com", "mastodon.example", etc.?

I see that many instances listed on https://joinmastodon.org/servers use the name "mastodon" in the domain name, but I can't find any place that confirms the legal use of the trademark.

### Motivation

I think that an official information should be provided somewhere (possibly in the repository source code), in order to avoid future issues.

**Comment by Bymiki** [comment]
I agree. Something to clarify as son as posible. We have the example of Wordpress, which restringed It too late and a lot of sites had to be rebranded: https://lorelle.wordpress.com/2006/10/26/using-wordpress-in-your-domain-name-dont/

**Comment by Gargron** [comment]
I’m working on bringing out a trademark usage policy as soon as possible!

**Comment by colinstu12** [comment]
Looks like this was implemented with #38548 ? 

For existing instances that contain 'mastodon' and 'mstdn' in their domains, should they be concerned about potential trademark suits being brought against them? 

Users have no way of fully migrating yet though? #12423

**Comment by kescherCode** [comment]
Naming your subdomain, or hell, perhaps even your domain "mastodon" or "mstdn" or "masto" is extremely common as-is. Many instances do this.
Heck, some instances *need to* put it in their local domain to differentiate it from other fedi software they are, or have been, running.

Looking beyond Mastodon, this is common practic with other fediverse software.
Looking beyond even that, this is extremely common practice with deployed software in general. People will name their subdomains, not just for their "web domains", and that's a good thing, actually.

Making a trademark policy go against this, as #38548 indicates (and as the policy itself does state since 2022), is a really, really bad move.
To me personally, it is a sign of closing down a common, and extremely petty.

**Comment by mailbun** [comment]
The way an software warning was added after 2 years since policy was enacted (see https://joinmastodon.org/trademark, `Last updated December 21, 2022` as of posting this) is also very concerning:

- For 2 years people were seemingly not meaningfully informed of this, and this policy was not enforced, so even if only post-policy-enactment violations are being counted for enforcement, it would lead to well-meaning people being affected.
- Setup process only warns about it during setup, meaning people may have already bought a domain at this point.

And I also think that this policy goes against the common practice, and is a massive shot in the foot with any practical benefits being outweighed by damage it will do, while compromise is very much possible (for example, allow usage of "mastodon" in instance names and domains as long as it follows some guidelines that would be needed avoid confusion and bad actors using the name).

**Comment by MadokVaur** [comment]
> The way an software warning was added after 2 years since policy was enacted (see https://joinmastodon.org/trademark, `Last updated December 21, 2022` as of posting this) is also very concerning:
> 
>     * For 2 years people were seemingly not meaningfully informed of this, and this policy was not enforced, so even if only post-policy-enactment violations are being counted for enforcement, it would lead to well-meaning people being affected.
> 
>     * Setup process only warns about it during setup, meaning people may have already bought a domain at this point.
> 
> 
> And I also think that this policy goes against the common practice, and is a massive shot in the foot with any practical benefits being outweighed by damage it will do, while compromise is very much possible (for example, allow usage of "mastodon" in instance names and domains as long as it follows some guidelines that would be needed avoid confusion and bad actors using the name).

Given the recent (at least to my knowledge) announcement of Andy Piper's resignation (into an ongoing Github thread, effective Apr 28 I believe) I'd say, with my usual snarky manner, that y'all maybe ain't seen nothin' yet

For those with a larger world experience, paying attention at the edges, it's always been apparent that while the lawyers have been lurking in the shadows only rarely have they stepped forward into the light and made themselves both visible and controlling

Here we go...

**Comment by mailbun** [comment]
Also, there is a related github discussion with more clarifications - https://github.com/mastodon/mastodon/discussions/22785

I am not sure where exactly should this issue be discussed, as I don't think it would be good to divide problems with how this is handled in instance setup and the underlying trademark policy problems, but this is an old issue that predates trademark policy itself.

**Comment by Doomsk** [comment]
does it make sense to "trademark" what a name is used as a subdomain (or even domain)? I mean, there's the Mastodon band... should they sue Mastodon the federated platform? Or should scientists sue both? And also, what if anyone wants to do anything with their domain names, doesn't seem to make any sense. Unless the website is being impersonating Mastodon's, it seems unreasonable.

**Comment by MadokVaur** [comment]
> does it make sense to "trademark" what a name is used as a subdomain (or even domain)? I mean, there's the Mastodon band... should they sue Mastodon the federated platform? Or should scientists sue both? And also, what if anyone wants to do anything with their domain names, doesn't seem to make any sense. Unless the website is being impersonating Mastodon's, it seems unreasonable.

OK: about to get off this and go back to watching Bluesky, which is being far more entertaining ATM...

I myself hold a US registered trademark for my photography business

I will say that, at least in the United States, the sub-category(-ies) to which a trademark applies are very narrow and very tightly controlled

Thus a trademark for 'photography' will not, by definition, infringe on a trademark using an identical word, but applied to 'snack foods'

I mean, of course you can get a lawyer to *try* anything, but...

*That* said, using "Mastodon" as a sub-domain for a Mastodon software distribution installation is likely problematic

IANAL &c &c &c

**Comment by Doomsk** [comment]
> > does it make sense to "trademark" what a name is used as a subdomain (or even domain)? I mean, there's the Mastodon band... should they sue Mastodon the federated platform? Or should scientists sue both? And also, what if anyone wants to do anything with their domain names, doesn't seem to make any sense. Unless the website is being impersonating Mastodon's, it seems unreasonable.
> 
> OK: about to get off this and go back to watching Bluesky, which is being far more entertaining ATM...
> 
> I myself hold a US registered trademark for my photography business
> 
> I will say that, at least in the United States, the sub-category(-ies) to which a trademark applies are very narrow and very tightly controlled
> 
> Thus a trademark for 'photography' will not, by definition, infringe on a trademark using an identical word, but applied to 'snack foods'
> 
> I mean, of course you can get a lawyer to _try_ anything, but...
> 
> _That_ said, using "Mastodon" as a sub-domain for a Mastodon software distribution installation is likely problematic
> 
> IANAL &c &c &c

Good points. But from what I understood, it's about instances (quoted from the main issue "i.e. independent websites that run an instance of this software"), not distribution or licensing of any software. To me It seems like Nike trying to sue some website for using a subdomain for selling their Nike products.

**Comment by retsplines** [comment]
+1 on this being a bit weird. Some nuance is needed. In my view:

Not okay: Straight up naming your site or other entity with `mastodon` as a substring of its primary domain/name. Examples (haven't checked if any of these are real/exist so, use caution): `mastodonspot.com`, `photographymastodon.com`, `artmastodon.com` 

Okay: Using `mastodon` to designate a service run on a subdomain of your broader site. Examples: `mastodon.example.com`, `mstdn.localhackspace.com`.

I am reminded of the [PHP License](https://www.php.net/license/3_0.txt) which carves something similar out specifically (emphasis mine):

>   4. Products derived from this software may not be called "PHP", nor may "PHP" appear in their name, without prior written permission from `group@php.net`.  **You may indicate that your software works in conjunction with PHP by saying "Foo for PHP" instead of calling it "PHP Foo" or "phpfoo"**

**Comment by mon-jai** [comment]
> Given the recent (at least to my knowledge) announcement of Andy Piper's resignation...

Where was this announced? I couldn't find it with search.

https://github.com/search?q=org%3Amastodon+commenter%3Aandypiper&type=issues

**Comment by MadokVaur** [comment]
> > Given the recent (at least to my knowledge) announcement of Andy Piper's resignation...
> 
> Where was this announced? I couldn't find it with search.
> 
> https://github.com/search?q=org%3Amastodon+commenter%3Aandypiper&type=issues

On Github, here: https://github.com/mastodon/mastodon/discussions/36552#discussioncomment-16576082

**Comment by YJSoft** [comment]
#40143 and https://blog.joinmastodon.org/2026/08/sharing-guidelines-about-mastodons-trade-mark-policy/ solves this issue.
